assets/scripts/dialogs/GeotagDialog.jsx

```diff
@@ -48,148 +44,138 @@ L.Icon.Default.mergeOptions({
   shadowUrl: '/images/marker-shadow.png'
 })
 
-/**
-This Dialog uses the generic Dialog component and combines it with the GeoSearch
-and LocationPopup components as well as a map to display the coordinates on
-It handles setting, displaying, and clearing location information assocaited with a 'street'
- */
-
-class GeotagDialog extends React.Component {
-  static propTypes = {
-    // Provided by react-intl higher-order component
-    intl: PropTypes.object.isRequired,
-
-    // Provided by Redux store
-    street: PropTypes.object,
-    markerLocation: PropTypes.shape({
-      lat: PropTypes.number,
-      lng: PropTypes.number
-    }),
-    addressInformation: PropTypes.object,
-    userLocation: PropTypes.shape({
-      latitude: PropTypes.number,
-      longitude: PropTypes.number
-    }),
-    dpi: PropTypes.number,
+GeotagDialog.propTypes = {
+  // Provided by Redux store
+  street: PropTypes.object,
+  markerLocation: PropTypes.shape({
+    lat: PropTypes.number,
+    lng: PropTypes.number
+  }),
+  addressInformation: PropTypes.object,
+  userLocation: PropTypes.shape({
+    latitude: PropTypes.number,
+    longitude: PropTypes.number
+  })
+}
 
-    // Provided by Redux action dispatchers
-    setMapState: PropTypes.func,
-    addLocation: PropTypes.func,
-    clearLocation: PropTypes.func,
-    saveStreetName: PropTypes.func
+function getInitialState (props) {
+  // Determine initial map center, and what to display
+  let mapCenter, zoom, markerLocation, label
+
+  // If street has a location object, use its position and data
+  if (props.street.location) {
+    mapCenter = props.street.location.latlng
```

Comment:
my instinct was to move markerLocation to be right after mapCenter so its more obvious they are the same value. 

however, i can see some semantic separation here in that the first two values are 'map' values, the second two are 'location' values, so I wanted to be mindful of that.